### DISCUSSION TITLE
Automations - Generating Multiple rows

### DISCUSSION THREAD

**Discussion by insomniacj93** [Budibase/budibase#7426] [Help] [answered]
Hello,

I'm having trouble with automations. I want to create an automation that generates multiple rows in a different table depending on the quantity stored or by user input (if instead of using for e.g. 3 (stored in table1), i just want to send 2 over to table2).

E.g.

table1

id - ProductA
quantity - 3

(table2 is empty)

App Action (automation)

=

table2

id - product1         id - product2         id - product3 
o_id - ProductA     o_id - ProductA     o_id - ProductA

table1

id - ProductA
quantity - 0

I'm using MySQL.

Thanks in advance!

**Comment by R2bEEaton** [comment, marked as answer]
Hi, @insomniacj93.

This is totally doable, using something called [Automation Looping](https://docs.budibase.com/docs/automation-actions#looping). Here's a short description for how to do what you want. Please see the accompanying export to see exactly how I did it.

1. Set up the tables like you had (in my example, Table 1 is called Stock and Table 2 is called Orders).
2. Make an automation.
3. Make it be activated by an app trigger, and add inputs for the product name and for an amount.
4. The first step should query rows in the Stock table with the filter set to `{{ trigger.fields.product_id }}` or whatever you named it. Limit this to 1. You can also use the actual `_id` of the product's row to ensure that only one result comes back, but as long as each product_id is unique, this should suffice.
5. Step 2 should be a Condition (so that users can't take out more product than you have): `{{ toInt trigger.fields.amount }}` Less Than `{{ add steps.1.rows.0.Quantity 1 }}`. The condition will continue if the amount to order is less than or equal to the quantity of stock.
6. Step 3 is a Create Row in the Orders table with the `product_id` set to `{{ trigger.fields.product_id }}`. You can also set an order number, but I just used the Auto ID column provided by Budibase.
7. _Add Looping to Step 3_. Set the input type to String and the Binding/Value to the following JavaScript:
```
var output = [...Array($("trigger.fields").amount + 1).keys()];
output.shift();
return output.join(",");
```
8. Finally, make a fourth step to Update Row for the Stock table. Set the quantity to the remaining quantity after subtracting the order `{{ subtract steps.1.rows.0.Quantity trigger.fields.amount }}` and make sure to address the correct Row ID with `{{ steps.1.rows.0._id }}`.

The Automation is done! Now all you need is a front-end! You can add some validation such that it won't let you submit the form unless you have an order that is less than or equal to the quantity, but ultimately the Automation will make sure that nothing goes negative.

A working example of this is attached here, along with a front-end. Note: Please refresh the page after each order in this example to see updated values.
[StockOrders-export-1661376564936.txt](https://github.com/Budibase/budibase/files/9419554/StockOrders-export-1661376564936.txt)

Reply if any of this is confusing or you need further help!

**Reply by insomniacj93** [reply]
Thank you!! I'll give it a try today!

**Reply by insomniacj93** [reply]
I encountered an error after running it a couple times.

I get "response": "Error: Duplicate entry 'productA' for key 'tb_Stock.PRIMARY'"

On Step 4, Updating Row, it appears to be trying to create a new row instead of updating and since the productA is defined to be a PK so i can use it around the other tables as a connection, is there a way to simply update the row instead of registering a new one?

Many thanks!

**Reply by R2bEEaton** [reply]
Hi @insomniacj93,

Can I see a screenshot of the automation steps? Particularly the trigger, the query, and the update row.
Of course, make sure that you pass the Row ID correctly with an existing `_id` so that it knows which row to update.
